ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py

```diff
@@ -143,27 +150,50 @@ def _get_backup_restore_connection_params(connection_options: dict) -> List[str]
     return params
 
 
-def _get_pg_exe_path(exe_name: str) -> str:
+def _get_pg_exe_path(exe_name: str, server_version: Tuple[int, int, int]) -> str:
     """
-    Find the path to the given PostgreSQL utility executable for the current operating system
+    Find the path to the given PostgreSQL utility executable for the current operating system in a server specific version folder
 
     :param exe_name: The name of the program to find (without .exe). e.g. 'pg_dump'
+    :param server_version: Tuple of the connected server version components (major, minor, ignored)
     :returns: The path to the requested executable
     :raises ValueError: if there is no file corresponding to the given exe_name
     """
+
     base_location = os.path.join(os.path.dirname(sys.argv[0]), 'pg_exes')
     platform = sys.platform
     if platform == 'win32':
-        path = os.path.join(base_location, 'win', exe_name + '.exe')
+        os_root = os.path.join(base_location, 'win')
+        path_suffix = exe_name + '.exe'
     elif platform == 'darwin':
-        path = os.path.join(base_location, 'mac', 'bin', exe_name)
+        os_root = os.path.join(base_location, 'mac')
+        path_suffix = os.path.join('bin', exe_name)
     else:
-        path = os.path.join(base_location, 'linux', 'bin', exe_name)
-
-    # Verify that the file exists
-    if not os.path.exists(path):
-        raise ValueError(f'Could not find executable file {path}')  # TODO: Localize
-    return path
+        os_root = os.path.join(base_location, 'linux')
+        path_suffix = os.path.join('bin', exe_name)
+
+    # Get the list of folders in the os specific root folder
+    all_folders: List[str] = [os.path.normpath(x[0]) for x in os.walk(os_root)]
+    for folder in all_folders:
+        folderName = os.path.basename(folder)
+        version = folderName.split('.')
+        # Get the major version value
+        try:
+            major = int(version[0])
```

Comment:
Shouldn't we do this assignment outside the for loop?  No need to run the conversion on each iteration, I think.